Add documentation for MELCloud integration (#11930)
home-assistant/home-assistant.github.io: source/_integrations/melcloud.markdown

---
title: MELCloud
description: MELCloud integration
logo: home-assistant.png
ha_category:
  - Climate

> ha_release: '0.106'

Reviewer: ```suggestion
ha_release: 0.106
```

ha_iot_class: Cloud Polling
ha_config_flow: true
ha_codeowners:
  - '@vilppuvuorinen'
---

> The `melcloud` integration integrates Mitsubishi Electric's MELCloud enabled devices into Home Assistant.

Reviewer 2: Is there some site that could be linked here?

Author: Linked.

## Device support

> - Air-to-Air heat pumps, i.e. AC units - **Supported**

Reviewer 2: do you mean "e.g." instead of "i.e." here?

Author: Not really, but it does look a lot better with "e.g.". Switched.

- Air-to-Water heat pumps - **Work in progress**
- Energy recovery ventilators - **Not supported**
- Other - **Not supported**

## Configuration

> MELCloud integration should be configured through the Home Assistant user interface using MELCloud  login details. This is not optimal, but **the provided password is not stored**.

Reviewer 2: ```suggestion
The integration should be configured through the user interface ("Configurations -> Integrations") using the MELCloud login details. While not optimal, **the provided password is not stored**.
```

> The stored access token can later be updated by adding the MELCloud integration again with the same email address.

Reviewer 2: Can or need to be updated?

Author: Attempted to clarify.

> Configuration is also possible through `configuration.yaml`. The token can be found from `X-MitsContextKey` header when logged into MELCloud with 'English' and 'Remember me'.

Reviewer 2: ```suggestion
Configuration is also possible through `configuration.yaml`. The required authentication token can be found in `X-MitsContextKey` header when logged into the MELCloud with 'English' and 'Remember me'.
```

Reviewer 2: "with 'English' and 'Remember me'" sounds odd. Is there are word missing or what's the message this is trying to convey? That it only works when the language is set to English and the "Remember me" option is selected while logging in?

Author: I repurposed the latter sentence.

```yaml
melcloud:
  token: xxxxxxxxxxxxxxxxxxx
```

{% configuration %}
token:
  description: X-MitsContextKey access token
  required: true
  type: string
{% endconfiguration %}

## Air-to-Air device

Air-to-Air heat pump provides `climate` and `sensor` platforms.

### Climate

> Following parameters can be controlled throught the `climate` platform:

Reviewer 2: ```suggestion
The following parameters can be controlled for the `climate` platform entities:
```

- Target temperature

> - Operation mode (HVAC mode) with emulated power off.

Reviewer 2: Is "emulated power off" understandable for end users?

Author: You'll never know. I tried to clarify this a bit.

- Fan speed

### Sensor

> Following parameters are provided through `sensor` platform:

Reviewer 2: ```suggestion
The following attributes are available for `sensor` platform entities:
```
